client-datastore-cassandra/src/main/java/com/telecominfraproject/wlan/client/datastore/cassandra/ClientSessionDAO.java

```diff
@@ -494,27 +494,24 @@ public ClientSession deleteSession(int customerId, long equipmentId, MacAddress
 		case customer_and_equipment:
 		case customer_and_location:
 		case customer_and_macAddress:
-			//the query was against client_session_by_equipment or client_session_by_location table
-			//find all the macAddresses for the page, then retrieve records for them from client_session table
-			Set<MacAddress> macAddrSet = new HashSet<>();
+			//the query was against client_session_by_equipment, client_session_by_location, or client_session_by_mac(_and_equipment) table
+			//find all the macAddresses and equipmentId for the page, then retrieve records for them from client_session table
 			while (rs.getAvailableWithoutFetching() > 0) {
-				macAddrSet.add(new MacAddress(rs.one().getLong("macAddress")));
-			}
-
-			//get all the sessions for the involved mac addresses
-			List<ClientSession> pageSessions = getSessions(customerId, macAddrSet);
-
-			//apply local filtering because retrieved sessions may be from different equipments and locations
-			pageSessions.forEach(cs -> {
-			    //apply locationId and equipmentId filtering in here
-                if ((locationIds == null || locationIds.isEmpty() || locationIds.contains(cs.getLocationId()))
-                        && (equipmentIds == null || equipmentIds.isEmpty() || equipmentIds.contains(cs.getEquipmentId()))
-                        && (macSubstring == null || macSubstring.isEmpty() || cs.getMacAddress().getAddressAsString().toLowerCase().contains(macSubstring.toLowerCase()))
-                		) 
+			    Row row = rs.one();
```

Comment:
Yes, the session retrieval filtering is the same, but now each filter table row will return 1 client session for the final response instead of each filter table row potentially return x number of client sessions.
For unit test, see the other comment